webpack/assets/javascripts/foreman_toast_notifications.js

```diff
@@ -13,15 +13,28 @@ export function notify({message, type, link, sticky = isSticky(type)}) {
       }));
 }
 
+export function clear() {
+  store.dispatch(ToastActions.clearToasts());
+}
+
 function importFlashMessagesFromRails() {
-  $('#notifications').data().flash.forEach(([type, message]) => {
+  const notifications = $('#notifications');
+
+  if (notifications.length === 0 ||
+    !notifications.data() ||
+    !notifications.data().flash) {return;}
```

Comment:
you can safely combine these two checks to `!notifications.data('flash')` (also works for looping below)